javascript/ql/src/semmle/javascript/Classes.qll

```diff
@@ -210,6 +217,23 @@ class ClassDeclStmt extends @classdeclstmt, ClassDefinition, Stmt {
  * A class expression.
  */
 class ClassExpr extends @classexpr, ClassDefinition, Expr {
+  override string getName() {
+    result = ClassDefinition.super.getName()
+    or
+    exists(VarDef vd | this = vd.getSource() | result = vd.getTarget().(VarRef).getName())
```

Comment:
Do we want to require `not exists(getIdentifier())` here and in the other cases below? Otherwise we might get multiple results.

Reply:
We're in danger of going in circles here, as @esben-semmle asked for multiple results, . Though I can see that it's more risky to do so now that it's merged into `getName()`.

Reply:
Exactly. I would expect `getName()` to have a single result.